### PR TITLE
integrate JSON file into single css theme builds

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What Changed

The previous PR would break the default (all themes) build and none of the theme json would match up leading to unstyled components. This change reworks that to let the build plugin handle the moduleHash and modifies the babel plugin to replace the imports to the `.css.js` file too.

Todo:

- [x] Add tests
- [x] Add docs

<details>
  <summary>📦 Published PR as canary version: <code>2.10.1-canary.603.13514.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @design-systems/babel-plugin-include-styles@2.10.1-canary.603.13514.0
  npm install @design-systems/babel-plugin-replace-styles@2.10.1-canary.603.13514.0
  npm install @design-systems/cli-utils@2.10.1-canary.603.13514.0
  npm install @design-systems/cli@2.10.1-canary.603.13514.0
  npm install @design-systems/core@2.10.1-canary.603.13514.0
  npm install @design-systems/create@2.10.1-canary.603.13514.0
  npm install @design-systems/docs@2.10.1-canary.603.13514.0
  npm install @design-systems/eslint-config@2.10.1-canary.603.13514.0
  npm install @design-systems/load-config@2.10.1-canary.603.13514.0
  npm install @design-systems/next-esm-css@2.10.1-canary.603.13514.0
  npm install @design-systems/plugin@2.10.1-canary.603.13514.0
  npm install @design-systems/stylelint-config@2.10.1-canary.603.13514.0
  npm install @design-systems/utils@2.10.1-canary.603.13514.0
  npm install @design-systems/build@2.10.1-canary.603.13514.0
  npm install @design-systems/bundle@2.10.1-canary.603.13514.0
  npm install @design-systems/clean@2.10.1-canary.603.13514.0
  npm install @design-systems/create-command@2.10.1-canary.603.13514.0
  npm install @design-systems/dev@2.10.1-canary.603.13514.0
  npm install @design-systems/lint@2.10.1-canary.603.13514.0
  npm install @design-systems/playroom@2.10.1-canary.603.13514.0
  npm install @design-systems/proof@2.10.1-canary.603.13514.0
  npm install @design-systems/size@2.10.1-canary.603.13514.0
  npm install @design-systems/storybook@2.10.1-canary.603.13514.0
  npm install @design-systems/test@2.10.1-canary.603.13514.0
  npm install @design-systems/update@2.10.1-canary.603.13514.0
  # or 
  yarn add @design-systems/babel-plugin-include-styles@2.10.1-canary.603.13514.0
  yarn add @design-systems/babel-plugin-replace-styles@2.10.1-canary.603.13514.0
  yarn add @design-systems/cli-utils@2.10.1-canary.603.13514.0
  yarn add @design-systems/cli@2.10.1-canary.603.13514.0
  yarn add @design-systems/core@2.10.1-canary.603.13514.0
  yarn add @design-systems/create@2.10.1-canary.603.13514.0
  yarn add @design-systems/docs@2.10.1-canary.603.13514.0
  yarn add @design-systems/eslint-config@2.10.1-canary.603.13514.0
  yarn add @design-systems/load-config@2.10.1-canary.603.13514.0
  yarn add @design-systems/next-esm-css@2.10.1-canary.603.13514.0
  yarn add @design-systems/plugin@2.10.1-canary.603.13514.0
  yarn add @design-systems/stylelint-config@2.10.1-canary.603.13514.0
  yarn add @design-systems/utils@2.10.1-canary.603.13514.0
  yarn add @design-systems/build@2.10.1-canary.603.13514.0
  yarn add @design-systems/bundle@2.10.1-canary.603.13514.0
  yarn add @design-systems/clean@2.10.1-canary.603.13514.0
  yarn add @design-systems/create-command@2.10.1-canary.603.13514.0
  yarn add @design-systems/dev@2.10.1-canary.603.13514.0
  yarn add @design-systems/lint@2.10.1-canary.603.13514.0
  yarn add @design-systems/playroom@2.10.1-canary.603.13514.0
  yarn add @design-systems/proof@2.10.1-canary.603.13514.0
  yarn add @design-systems/size@2.10.1-canary.603.13514.0
  yarn add @design-systems/storybook@2.10.1-canary.603.13514.0
  yarn add @design-systems/test@2.10.1-canary.603.13514.0
  yarn add @design-systems/update@2.10.1-canary.603.13514.0
  ```
</details>

### CODE REVIEW

File: packages/babel-plugin-replace-styles/src/index.ts [hunk 1]
```diff
@@ -26,26 +26,45 @@ export default function replaceStyles(
...
+    state: PluginPass,
+    importName: string
+  ) {
+    const isScope =
+      state?.file?.opts?.filename &&
+      state.file.opts.filename.includes(`@${scope}/`);
+
+    if (isScope && path.node.specifiers.length) {
+      // eslint-disable-next-line no-param-reassign
+      path.node.source.value = importName.replace('.css', `-${use}.css`)
```
Comment (tylerkrupicka) [inline packages/babel-plugin-replace-styles/src/index.ts:67]: Should this also check if the file exists before swapping, like the CSS one?

**Review by tylerkrupicka — COMMENTED** [review]
Looks great!

File: packages/babel-plugin-replace-styles/src/index.ts (hunk 1 above)
Comment (hipstersmoothie) [reply packages/babel-plugin-replace-styles/src/index.ts:67]: good idea

**Comment by hipstersmoothie** [comment]
:rocket: PR was released in [`v2.10.1`](https://github.com/intuit/design-systems-cli/releases/tag/v2.10.1) :rocket:

**Pull request merged**